src/main/java/org/sunbird/ratings/service/RatingServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+            request.put(Constants.ACTIVITY_ID, requestRating.getActivity_Id());
+            request.put(Constants.ACTIVITY_TYPE, requestRating.getActivity_type());
+            request.put(Constants.RATINGS_USER_ID, requestRating.getUserId());
+
+
+            List<Map<String, Object>> existingDataList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
+                    Constants.TABLE_RATINGS, request, null);
+
+            if (!existingDataList.isEmpty()) {
+
+                Map<String, Object> updateRequest = new HashMap<>();
+                updateRequest.put(Constants.RATING, requestRating.getRating());
+                updateRequest.put(Constants.REVIEW, requestRating.getReview());
+                updateRequest.put(Constants.UPDATED_ON, timeBasedUuid);
+
+                Map<String, Object> prevInfo = existingDataList.get(0);
+                cassandraOperation.updateRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_RATINGS, updateRequest,
+                        request);
+
+                msg.setAction("ratingUpdate");
+                msg.setActivity_id(requestRating.getActivity_Id());
+                msg.setActivity_Type(requestRating.getActivity_type());
+                msg.setUser_id(requestRating.getUserId());
+                msg.setCreated_Date(String.valueOf((prevInfo.get("createdon"))));
+
+                UpdatedValues prevValues = processEventMessage(String.valueOf(prevInfo.get("createdon")),
+                        (Float) prevInfo.get("rating"), (String) prevInfo.get("review"));
+
+                UpdatedValues updatedValues = processEventMessage(String.valueOf(updateRequest.get(Constants.UPDATED_ON)),
+                        requestRating.getRating(), requestRating.getReview());
+
+                msg.setPrevValues(prevValues);
+                msg.setUpdatedValues(updatedValues);
+
+                producer.push(Constants.RATINGS_UPDATE_EVENT, msg);
+                response.getParams().setStatus(Constants.SUCCESSFUL);
+                response.setResponseCode(HttpStatus.CREATED);
+                response.setResponseCode(HttpStatus.OK);
+                return response;
+            }
+
+            request.put(Constants.CREATED_ON, timeBasedUuid);
+            request.put(Constants.RATING, requestRating.getRating());
+            request.put(Constants.REVIEW, requestRating.getReview());
+            request.put(Constants.UPDATED_ON, timeBasedUuid);
+
+            cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_RATINGS, request);
+
+            response.getParams().setStatus(Constants.SUCCESSFUL);
+            response.put(Constants.DATA, request);
+            response.setResponseCode(HttpStatus.CREATED);
+        } catch (ValidateRatingException ex) {
+            String errMsg = "Exception is : ";
+            return processExceptionBody(response, ex, errMsg, HttpStatus.BAD_REQUEST);
+        } catch (Exception ex) {
```

Comment:
We should not have a catch block for Exception which is very broad.

Reply:
as discussed leaving it as it can catch other cassandra or unexpected sorts of errors